pyvista/core/datasetattributes.py

```diff
@@ -170,94 +266,243 @@ def get_array(self, key: Union[str, int]) -> Union[pyvista_ndarray, _vtk.vtkData
+    def set_array(self, data: Union[Sequence[Number], Number, np.ndarray],
+                  name: str, deep_copy=False, active_vectors=True,
+                  active_scalars=True) -> None:
         """Add an array to this object.
 
         Parameters
         ----------
-        narray : array_like, scalar value
+        data : sequence
             A ``pyvista_ndarray``, ``numpy.ndarray``, ``list``,
             ``tuple`` or scalar value.
 
         name : str
             Name of the array to add.
 
-        deep_copy : bool
+        deep_copy : bool, optional
             When ``True`` makes a full copy of the array.
 
-        active_vectors : bool
-            If ``True``, make this the active vector array.
+        active_vectors : bool, optional
+            If ``True``, also make this the active vector array.
+
+        active_scalars : bool, optional
+            If ``True``, also make this the active scalar array.
+
+        Examples
+        --------
+        Add a point array to a mesh without making it the active
+        scalars.
+
+        >>> import pyvista
+        >>> mesh = pyvista.Cube().clean()
+        >>> data = range(mesh.n_points)
+        >>> mesh.point_arrays.set_array(data, 'my-data', active_scalars=False)
+        >>> mesh.point_arrays['my-data']
+        pyvista_ndarray([0, 1, 2, 3, 4, 5, 6, 7])
+
+        Add a cell array to a mesh without making it the active
+        scalars.
+
+        >>> cell_data = range(mesh.n_cells)
+        >>> mesh.cell_arrays.set_array(cell_data, 'my-data',
+        ...                            active_scalars=False)
+        >>> mesh.cell_arrays['my-data']
+        pyvista_ndarray([0, 1, 2, 3, 4, 5])
+
+        Add a field array to a mesh.
+
+        >>> field_data = range(3)
+        >>> mesh.field_arrays.set_array(field_data, 'my-data')
+        >>> mesh.field_arrays['my-data']
+        pyvista_ndarray([0, 1, 2])
+
+        Notes
+        -----
+        You can simply use the ``[]`` operator to add an array to the
+        point, cell, or field data.  Note that by default if this is
+        not field data, the array will be made the active scalars, and
+        if a ``(N x 3)`` shaped array, then active vector data as
+        well.
 
-        active_scalars : bool
-            If ``True``, make this the active scalar array.
         """
-        if narray is None:
-            raise TypeError('narray cannot be None.')
-        if isinstance(narray, Iterable):
-            narray = pyvista_ndarray(narray)
+        vtk_arr, shape = self._prepare_array(data, name, deep_copy)
+        self.VTKObject.AddArray(vtk_arr)
+
+        try:
+            if active_scalars:
+                self.active_scalars = name  # type: ignore
+            if active_vectors:
+                # verify this is vector data
+                if len(shape) == 2 and shape[1] == 3:
+                    self.active_vectors = name  # type: ignore
+                    self.VTKObject.SetVectors(vtk_arr)
```

Comment:
I think this functionality of setting either scalar or vector, should be separated from whether it is set as the active scalar or vector.

There could be a keyword arg to choose, but it gets messy with the choices of active_scalars and active_vectors.  Can a vector data be set as active_scalars?

Reply:
Resolved in #1593.